File: python/day5_part2_2015.py
with open("input.txt", "r") as file:
    strings = file.read().splitlines()

nice_count = 0

for string in strings:
    if any([string.count(char_pair) >= 2 for char_pair in [string[i:i+2] for i in range(len(string)-1)]]):
        if any([string[i] == string[i+2] for i in range(len(string)-2)]):
            nice_count += 1

print(nice_count)

File: day5_part2_2015.rs

use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");
    
    let nice_count = input.lines().filter(|line| {
        let mut has_pair = false;
        let mut has_repeat_with_gap = false;

        for i in 0..line.len() - 2 {
            let pair = &line[i..i+2];
            if line[i+2..].contains(pair) {
                has_pair = true;
                break;
            }
        }

        for i in 0..line.len() - 2 {
            if line.chars().nth(i) == line.chars().nth(i + 2) {
                has_repeat_with_gap = true;
                break;
            }
        }

        has_pair && has_repeat_with_gap
    }).count();

    println!("{}", nice_count);
}
